auction-server/src/opportunity/repository/add_opportunity.rs

```diff
@@ -63,4 +70,42 @@ impl<T: InMemoryStore> Repository<T> {
 
         Ok(opportunity)
     }
+
+    async fn check_db_duplicate_opportunity(
+        &self,
+        db: &sqlx::Pool<Postgres>,
+        opportunity: &<T as InMemoryStore>::Opportunity,
+    ) -> Result<bool, RestError> {
+        let chain_type = <T::Opportunity as entities::Opportunity>::ModelMetadata::get_chain_type();
+        let metadata = opportunity.get_models_metadata();
+
+        let result = sqlx::query!("SELECT COUNT(*) FROM opportunity WHERE permission_key = $1 AND chain_id = $2 AND chain_type = $3 AND sell_tokens = $4 AND buy_tokens = $5 AND metadata #- '{slot}' = $6",
```

Comment:
I think we can ignore slot and we don't need to check the slot value.

Reply:
We remove opportunity when it's invalid, or expired.
I think if an opportunity is invalid, then we need to add a new row to the db. When the opportunity is expired, we can update the row and remove the removal time and consider this opportunity as a live opportunity again.